internal/controller/clusterpolicy_controller.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgedb/edgedb-go"
+	"github.com/pingcap/errors"
+
+	kyvernoV1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/policy-meta-operator/internal/utils"
+	edgedbutils "github.com/giantswarm/policy-meta-operator/internal/utils/edgedb"
+)
+
+// PolicyReconciler reconciles a AutomatedException object
+type ClusterPolicyReconciler struct {
+	client.Client
+	EdgeDBClient     *edgedb.Client
+	Scheme           *runtime.Scheme
+	MaxJitterPercent int
+}
+
+var (
+	//GiantSwarm team label
+	GSTeamLabel = "application.giantswarm.io/team"
+	//Policy API Exemption label
+	//TODO: Move to Policy API
+	PolicyAPIExemptionLabel = "policy.giantswarm.io/giantswarm-exempt"
+)
+
+func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	var clusterPolicy kyvernoV1.ClusterPolicy
+
+	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
+		if !errors.IsNotFound(err) {
+			// Error fetching the report
+			log.Log.Error(err, "unable to fetch Kyverno ClusterPolicy")
+		} else {
+			// Policy not found, make sure we don't have it in edgedb either
+			err := edgedbutils.DeleteKyvernoClusterPolicy(ctx, r.EdgeDBClient, req.Name)
+			if err != nil {
+				log.Log.Error(err, "Error deleting ClusterPolicy from database")
+			}
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if clusterPolicy.Spec.ValidationFailureAction.Enforce() {
```

Comment:
I think there's still some work to be done to ensure edgedb follows the etcd state.
For example, this would break if a policy is in `Enforce` and then switches to `Audit`, since this reconciliation would be skipped

Reply:
I want to add a delete case when Audit is set, but if we are going to eventually have everything in edgedb I could just skip this point and not care about Enforced or Audit